ToDoList/test/ToDoList.API.Tests/Controllers/ToDosControllerTests.cs

```diff
@@ -0,0 +1,126 @@
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDosAsync());
+            response.TryGetContentValue(out IEnumerable<ToDo> result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(_toDoList).UsingToDoComparer(), "Todos are not equal");
+        }
+
+        [Test]
+        public async Task GetToDoAsync_CorrectToDoReturned()
+        {
+            // Arrange
+            const int itemIndex = 0;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+            _toDoRepositorySubstitute.GetToDoAsync(expectedToDo.Id).Returns(expectedToDo);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDoAsync(expectedToDo.Id));
+            response.TryGetContentValue(out ToDo result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(expectedToDo).UsingToDoComparer(), $"{result} is not equal to expected {expectedToDo}");
+        }
+
+        [Test]
+        public async Task PostToDoAsync_IsAdded_NewToDoWithEqualRouteReturned()
+        {
+            // Arrange
+            const int itemIndex = 2;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+
+            _toDoRepositorySubstitute.AddToDoAsync(expectedToDo).Returns(expectedToDo);
+            _urlLocationServiceSubstitute.GetNewResourceLocation(expectedToDo.Id).Returns($"todos/{expectedToDo.Id}");
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.PostToDoAsync(expectedToDo));
+            response.TryGetContentValue(out ToDo result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.Created), $"Expecting status code Created, but was {response.StatusCode}");
+            Assert.That(response.Headers.Location.ToString(), Is.EqualTo($"todos/{expectedToDo.Id}"), $"Location of new todo is not as expected, was {response.Headers.Location}");
+            Assert.That(result, Is.EqualTo(expectedToDo).UsingToDoComparer(), $"{result} is not equal to expected {expectedToDo}");
+        }
+
+        [Test]
+        public async Task PutToDoAsync_IsChanged_NoContentReturned()
+        {
+            // Arrange
+            const int itemIndex = 2;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.PutToDoAsync(expectedToDo.Id, expectedToDo));
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.NoContent), $"Expecting status code NoContent, but is {response.StatusCode}");
```

Comment:
Hm, thinking about it, since you return `NoContent`, you should at least check that the `repository` method was called with correct parameters… (same applies to the `Delete` method)